docs/DeveloperGuide.md

```diff
@@ -426,6 +432,36 @@ An instance of `CommandResult` is then created which contains the message and in
    Applicants matching at least one keyword will be returned (i.e. OR search). 
    e.g. `find Ben Bobby` will return Ben Yang, Bobby Chin.
 
+### Design considerations
+
+#### Aspect: How to find applicants
+
+- Alternative 1: Find applicants by name using exact match search 
+(e.g. `find Eddy` will only result in applicants whose name is Eddy)
+  - Pros: Users can find applicants by their exact name, 
+  allowing for a more specific search
+  - Cons: Users would have to type in the entire name exactly in order to
+  get their desired result. This can result in higher user error and takes more time to type.
+  
+- Alternative 2 (current choice): Find applicants by name using partial match search (or "fuzzy" search)
+  - Pros: More inclusive, can find matches that are related but not exactly the same as `KEYWORD`.
+  More user-friendly as it accounts for variations or common misspellings. Allows for faster typing as users
```

Comment:
idk if our implementation really accounts for common misspellings, since it just searches for a substring right? I think the advantage is more of not having to type the full name of the applicant?